mindmeld/components/dialogue.py

```diff
@@ -689,6 +691,201 @@ def _get_dialogue_handler(self, dialogue_state):
         return handler
 
 
+class AutoEntityFilling(DialogueFlow):
+    """A special dialogue flow sublcass to implement Automatic Entitiy (Slot) Filling
+    (AEF) that allows developers to prompt users for completing the missing
+    requirements for entity slots.
+
+    Attributes:
+        app (Application): The application that initializes this flow.        
+    """
+
+    logger = mod_logger.getChild("AutoEntityFilling")
+    """Class logger."""
+
+    def __init__(self, name, entrance_handler, app, **kwargs):
+        super().__init__(name, entrance_handler, app)
+
+        self.app = app
+        self.entrance_handler = entrance_handler
+
+    def _set_target_state(self):
+        """Set target dialogue state to current flow"""
+        responder.params.target_dialogue_state = self.flow_state
+
+    def _exit_flow(self):
+        """Exits this flow and clears the related parameter for re-usability"""
+        if 'slot_not_prompted' in responder.frame:
+            del responder.frame['slot_not_prompted']
+        responder.exit_flow()
+
+    def _extract_query_features(self, text):
+        """ Extracts Query object from the user input and converts it into
+        appropriate format for entity extraction.
+
+        Args:
+            text (Request.text): Text in the query
+
+        Returns:
+            Formatted Payload (tuple(query (Query Object), list of entities, entity index))
+        """
+        resource_loader = self.app.app_manager.nlp.resource_loader
+
+        """ Extracting the Query object associated with the user input here.
+
+        process_markup args:
+            text, query factory, query options (optional dict) {language, time_zone and time_stamp}
+
+        returns: 
+            tuple(raw_text (str), Query object, list of entities)
+        """
+        query = process_markup(text, resource_loader.query_factory, {})[1]
```

Comment:
why the index `1`? Maybe assign `index = 1` and explain what index is doing

Reply:
@vijay120 It is noted in the docstring above that the return value of process_markup is `tuple(raw_text (str), Query object, list of entities)`, so index 1 is getting the Query object. Do you think it needs additional documentation?

Reply:
Added another line in the docstring to clarify this.

Reply:
Should we add the language/timezone/time_stamp as well?